### PR TITLE
feat(cli): support backend app for "app init"

### PR DESCRIPTION
**Pull request by efekarakus**
_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/cli/app_init.go [hunk 1]
```diff
@@ -229,7 +233,24 @@ func (o *initAppOpts) createManifest() (string, error) {
 	return manifestPath, nil
 }
 
-func (o *initAppOpts) createLoadBalancedAppManifest() (*manifest.LoadBalancedWebApp, error) {
+func (o *initAppOpts) newManifest() (encoding.BinaryMarshaler, error) {
+	switch o.AppType {
+	case manifest.LoadBalancedWebApplication:
+		return o.newLoadBalancedWebAppManifest()
+	case manifest.BackendApplication:
+		return manifest.NewBackendApp(manifest.BackendAppProps{
```
Comment (iamhopaul123) [inline internal/pkg/cli/app_init.go:241]: Would it be better to have `newBackAppManifest()` to be consistent with `newLoadBalancedWebAppManifest()`?

Comment (efekarakus) [reply internal/pkg/cli/app_init.go:241]: like have `o.newBackendAppManifest` that calls the constructor?

Comment (iamhopaul123) [reply internal/pkg/cli/app_init.go:241]: Yeah just a nit though

File: internal/pkg/cli/app_init.go [hunk 2]
```diff
@@ -24,9 +25,12 @@ import (
...
 var (
-	appInitAppTypePrompt     = "Which type of " + color.Emphasize("infrastructure pattern") + " best represents your application?"
-	appInitAppTypeHelpPrompt = `Your application's architecture. Most applications need additional AWS resources to run.
-To help setup the infrastructure resources, select what "kind" or "type" of application you want to build.`
+	appInitAppTypePrompt        = "Which type of " + color.Emphasize("infrastructure pattern") + " best represents your application?"
+	fmtAppInitAppTypeHelpPrompt = `A %s is a public, internet-facing, HTTP server that's behind a load balancer. 
+To learn more see: https://git.io/JfIpv
+
+A %s is a private, non internet-facing service.
+To learn more see: https://git.io/JfIpT`
```
Comment (seongm-1) [inline internal/pkg/cli/app_init.go:33]: Is the ` on purpose?

**Pull request merged**